lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -1261,12 +1257,11 @@ namespace Js
         JavascriptString* newString = nullptr;
         const char16* inputStr = input->GetString();
         CharCount inputLength = input->GetLength();
-        const int rawNumGroups = pattern->NumGroups();
+        const uint16 numGroups = pattern->NumGroups();
         Var nonMatchValue = NonMatchValue(scriptContext, false);
         UnifiedRegex::GroupInfo lastMatch; // initially undefined
 
-        AssertOrFailFast(0 < rawNumGroups && rawNumGroups <= INT16_MAX);
```

Comment:
Do we now allow numGroups to be zero? Looks like that was previously disallowed in this assertion.

Reply:
As best I remember, `numGroups` is always positive here, because the entire regexp counts as a capturing group.  The method can handle `numGroups` being zero, so this should be safe -- but I presumably had some reason for making that a strict inequality when I added that assertion as part of the MSRC.  I'll go ahead and restore it.